File: M/3127-make-a-square-with-the-same-color/main.go
func canMakeSquare(grid [][]byte) bool {
	for r := 1; r < 3; r++ {
		for c := 1; c < 3; c++ {
			b := 0
			w := 0
			for _, cord := range [][]int{{-1, -1}, {-1, 0}, {0, -1}, {0, 0}} {
				if grid[r+cord[0]][c+cord[1]] == 'B' {
					b++
				} else {
					w++
				}
			}
			if b <= 1 || w <= 1 {
				return true
			}
		}
	}
	return false
}

File: M/3127-make-a-square-with-the-same-color/main.rs
struct Solution {}

impl Solution {
  pub fn can_make_square(grid: Vec<Vec<char>>) -> bool {
    (1..3)
      .map(|r| {
        (1..3)
          .map(|c| {
            let v = if grid[r][c] == 'B' { 1 } else { 0 }
              + if grid[r - 1][c] == 'B' { 1 } else { 0 }
              + if grid[r][c - 1] == 'B' { 1 } else { 0 }
              + if grid[r - 1][c - 1] == 'B' { 1 } else { 0 };
            v <= 1 || v >= 3
          })
          .any((|x| x))
      })
      .any(|x| x)
  }
}
